eth/vm/tracing.py

```diff
@@ -0,0 +1,246 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+
+import contextlib
+
+import logging
+
+from typing import (
+    Dict,
+    Iterator,
+    NamedTuple,
+    Optional,
+    Tuple,
+    TYPE_CHECKING,
+    Union,
+)
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+
+from eth_utils import (
+    decode_hex,
+    ValidationError,
+)
+
+from eth.exceptions import (
+    VMError,
+)
+
+from eth.vm.opcode import (
+    Opcode,
+)
+
+if TYPE_CHECKING:
+    # avoid circular import
+    from eth.vm.computation import BaseComputation  # noqa: F401
+    from trinity.chains.base import BaseAsyncChain   # noqa: F401
```

Comment:
This is why the lint is failing: the code imports the `trinity` module from inside `eth`. Not sure the right solution here. I can't think of a reason off-hand why the `BaseAsyncChain` couldn't live in `eth` but I don't know why it was moved out in the first place...

Reply:
See my comment below about moving the utility (and thus not needing this import)